src/nodes.rs

```diff
@@ -135,17 +135,87 @@ impl<'a, T> IntoIterator for &'a NodeMap<T> {
     }
 }
 
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct BoolNodeMap(bit_vec::BitVec<u32>);
+
+impl BoolNodeMap {
+    pub fn new_with_len(len: NodeCount) -> Self {
+        BoolNodeMap(bit_vec::BitVec::from_elem(len.0, false))
+    }
+
+    pub fn set(&mut self, i: NodeIndex, value: bool) {
+        self.0.set(i.0, value);
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = bool> + '_ {
+        self.0.iter()
+    }
+
+    pub fn true_indices(&self) -> impl Iterator<Item = NodeIndex> + '_ {
```

Comment:
how about `node_indices` or just `nodes`?

Reply:
I wanted to make it compatible with the interface of NodeMap<T>. I also considered renaming BoolNodeMap to NodeSet, but set does not suggest existence of the total count of the nodes. Also, the len function would be very confusing. And technically, the false'y nodes are still in the BoolNodeMap, but with false value ...
Maybe the best solution would be to indeed rename the struct to NodeSet...

Reply:
Maybe NodeSet with capacity instead of length?

Reply:
Now that I think about it, there is no need for a capacity, we can simply define the capacity as i/8 + 1where i is the largest contained index (unless we really need to store the total number of nodes in the NodeSet, which seems a code smell)

Reply:
In a scenario where NodeSet is used it's the only data_structure that carries the info on total number of parents. Maybe it doesn't fit to this abstraction but is needed in the application. For me, personally, NodeSet with capacity is a valid construct. I would even propose renaming `new_with_len` to `new_with_capacity` as it seems more natural when I think about it now.

Reply:
`new_with_capacity` sounds good. But I think, we usually have the access to the key(box|chain), which seems a more appropriate place to store the node count?

Reply:
We use `n_members` from ControlHash in Terminal where we don't have access to the KeyBox. If you have an idea how to cleanly replace it with sth else, then the node count could disapear from NodeSet.

Reply:
I think capacity is a good name, and `new_with_capacity` is the way to go. Having capacity != n_members might backfire at some point, as the code is written with this assumption in mind (at least the part for representing unit parents).

Reply:
I will name it `with_capacity` as in `Vec::with_capacity`

Reply:
For the sake of Terminal I think it would not be a big deal to add a field n_members to Terminal. But generally the concern with not having fixed capacity are strange issues with encoding/hashing/validating units. We save a lot of effort (thinking about what the validation logic should be, what are the edge cases etc.) by specifying that parents map has fixed capacity.

Reply:
Agreed. The BoolNodeMap has now capacity.